enzyme/tools/enzyme-tblgen/datastructures.h

```diff
@@ -12,29 +16,24 @@
 
 using namespace llvm;
 
-enum argType { fp, len, vincData, vincInc };
-
-class Arg {
-  public:
-    size_t pos;
-    std::string name;
+enum argType {
```

Comment:
```suggestion
enum class argType {
```
enum classes are preferred because their values do not implicitly convert to other types (like another enum or int)

Reply:
TIL, thanks.